core/src/main/java/io/grpc/Deadline.java

```diff
@@ -141,4 +141,14 @@ public long timeRemaining(TimeUnit unit) {
   public String toString() {
     return timeRemaining(TimeUnit.NANOSECONDS) + " ns from now";
   }
+
+  @Override
+  public int compareTo(Deadline that) {
+    if (this.deadlineNanos < that.deadlineNanos) {
```

Comment:
This is incorrect since nanoTime() is permitted to loop such that overflow/underflow don't apply in the normal way. Instead, you should subtract one from the other and return based on whether the result is positive, negative, or zero.